java/core/src/java/org/apache/orc/impl/RunLengthIntegerWriterV2.java

```diff
@@ -286,7 +286,10 @@ private void writePatchedBaseValues() throws IOException {
     // find the number of bytes required for base and shift it by 5 bits
     // to accommodate patch width. The additional bit is used to store the sign
     // of the base value.
-    final int baseWidth = utils.findClosestNumBits(min) + 1;
+    int baseWidth = utils.findClosestNumBits(min) + 1;
```

Comment:
Same comment applies here (as the C++ version) -- I would try to avoid the extra if statement.

Can we please add some java tests as well?

Reply:
OK, I add this java tests

Reply:
OK, I add java test into the pull request.